README.md
Angular Spidergraph - Dynamic, interactive spidergraphs in HTML5
========================================================================

Angular spidergraph is a directive library module that creates nice looking, interactive spidergraphs in HTML5, using the canvas element.
It is using https://github.com/jmstriegel/jquery.spidergraph internally.
For more information visit https://github.com/jmstriegel/jquery.spidergraph

What are spidergraphs good for?
---------------------------------

- illustrating scaled quantitative data for several subjective attributes
- overlaying multiple data measurements for attribute comparison
- visualizing the intersection of several data measurements


How do I use it?
----------------

Minimum use:
----------------
Include https://github.com/jmstriegel/jquery.spidergraph jquery spider graph then 

Download it from here or download via bower 
```
    bower install angular-spider-graph
```
angular spider graph

Include the module in your app like 
```
    angular.module('myApp', ['morwal.spiderGraph'])
```
First make a div to contain your graph:
```
    <spider-map labels="labels" ng-model="activeData" active-data="activeData"></spider-map>
```
in your controller set active data
```
        $scope.labels =[ "Designing", "Coding", "Cycling", "Running"];
        $scope.activeData = [0,1,4,0];
```

There is two way binding in model activeData
Thats it.

Advanced uses
----------------
Change grid color 
in controller 
```
    $scope.gridColor='rgba(20,20,20,0.2)'
```
In view 
```
    <spider-map labels="labels" ng-model="activeData" active-data="activeData" grid-color="gridColor"></spider-map>
```

Changing configurtion for active data=
```
  $scope.activeConfig={
 'strokecolor': 'rgba(0,204,230,0.8)',
 'fillcolor': 'rgba(0,204,230,0.6)',
 'linear':true
 };
```
 It is options to set these parameters. linear is default true. You can set it to false on need. 
```
 <spider-map labels="labels" ng-model="activeData" active-data="activeData" active-config="activeConfig"></spider-map>
```
Add static layers to your data set. Opacity can be used to see through multiple layers.
```
    <spider-map labels="labels" ng-model="activeData" active-data="activeData" grid-color="gridColor" layers="layers"></spider-map>
```
This model will be array. Here only data is complusary paramater.
```
$scope.layers=[
 {
 'strokecolor': 'rgba(230,204,0,0.8)',
 'fillcolor': 'rgba(230,204,0,0.6)',
 'data': [5, 4, 9, 8, 1]
 }
 ];
```

Can I see an example?
-----------------------

Yes you can. Right over here:

Who created this
-----------------------
Shankar Morwal (https://github.com/morwalz)


CHANGELOG.md
Change Log
========================================================================

V 0.1
----------------
First release



Gruntfile.js
module.exports = function(grunt) {
    // load all grunt tasks from package.json matching the 'grunt-*' pattern
    require('load-grunt-tasks')(grunt);

    // Project configuration.
    grunt.initConfig({
        distdir: 'build',
        pkg: grunt.file.readJSON('package.json'),

        // Task configuration.
        clean: {
            dist: ['<%= distdir %>/*']
        },

        copy: {
            assets: {
                files: [
                    { dest: '<%= distdir %>', src: '**', expand: true, cwd: 'src/' }
                ]
            }

        },
        uglify: {
            dist: {
                src:['<%= distdir %>/**/*.js'],
                dest: '<%= distdir %>/angular-spider-graph.min.js'
            }
        }
    });

    // Task registration
    grunt.registerTask('default', ['clean','copy','uglify']);
};


src/angular-spider-graph.js
/*jshint multistr:true, curly: false */
/*global jQuery:false, define: false */
/**
 * Angular spider graph
 *
 * Written by
 * ----------
 * Shankar Morwal (morwal89@gmail.com,https://github.com/morwalz)
 *
 * Licensed under the MIT (MIT-LICENSE.txt).
 *
 * @author Shankar Morwal
 * @version 0.1-RELEASE
 *
 * Dependencies
 * ------------
 * jQuery (http://jquery.com)
 * jQuery spidergraph(https://github.com/jmstriegel/jquery.spidergraph)
 * Angular.js
 *
 **/
;
angular.module("morwal.spiderGraph", [])
  .directive('spiderMap', function() {
    return {
      restrict: 'AE',
      scope: {
        labels:'=',
        activeData:'=',
        activeConfig:'=',
        layers:'=',
        gridColor:'='
      },
      replace: true,
      require: 'ngModel',
      transclude: true,
      template: '<div id="spidergraphcontainer"></div>',
      link: function(scope, elem, attrs,ngModelCtrl) {
        var gridColor=(scope.gridColor)?scope.gridColor:'rgba(20,20,20,0.2)';
        $(elem).spidergraph({
          'fields':scope.labels,
          'gridcolor':scope.gridColor
        });
        var activeData;
        if(scope.activeConfig){
          activeData={
            'strokecolor':(scope.activeConfig.strokecolor)?scope.activeConfig.strokecolor:'rgba(102,195,207,0.8)',
            'fillcolor': (scope.activeConfig.fillcolor)?scope.activeConfig.fillcolor:'rgba(160,224,232,0.6)',
            'data':scope.activeData,
            'linear':(scope.activeConfig.linear===false)?scope.activeConfig.linear:true
          }
        }else{
          activeData={
            'strokecolor':'rgba(102,195,207,0.8)',
            'fillcolor': 'rgba(160,224,232,0.6)',
            'data':scope.activeData,
            'linear':true
          }
        }
        $(elem).spidergraph('setactivedata',activeData);
        if(scope.layers){
          scope.layers.forEach(function(layer){
            $(elem).spidergraph('addlayer', {
              'strokecolor':(layer.strokecolor)?layer.strokecolor:'rgba(102,195,207,0.8)',
              'fillcolor':(layer.fillcolor)?layer.fillcolor:'rgba(160,224,232,0.6)',
              'data':layer.data,
              'linear':(layer.linear)?layer.linear:true
            });
          })
        }
        scope.$watch('activeData', function (newValue, oldValue) {
          $('#spidergraphcontainer').spidergraph('redraw');
        },true);
        $(elem).bind('spiderdatachange', function( ev, data ) {
          scope.$apply(function () {
            ngModelCtrl.$setViewValue(data);
          });
          scope.activeData=data;
        });
      }
    };
  });